k8s/pkg/schedulers/scheduler_outside_cluster.go

```diff
@@ -0,0 +1,112 @@
+/** Copyright 2020-2023 Alibaba Group Holding Limited.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package schedulers implements the vineyard scheduler plugin.
+package schedulers
+
+import (
+	"strconv"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// VineyardSchedulerOutsideCluster is the vineyard scheduler outside cluster
+type VineyardSchedulerOutsideCluster struct {
+	client.Client
+	annotations map[string]string
+	labels      map[string]string
+	config      SchedulerConfig
+}
+
+// NewVineyardSchedulerOutsideCluster returns a new vineyard scheduler outside cluster
+func NewVineyardSchedulerOutsideCluster(
+	c client.Client,
+	annotations map[string]string,
+	labels map[string]string,
+	namespace string,
+	ownerReferences []metav1.OwnerReference,
+) *VineyardSchedulerOutsideCluster {
+	return &VineyardSchedulerOutsideCluster{
+		Client:      c,
+		annotations: annotations,
+		labels:      labels,
+		config: SchedulerConfig{
+			Namespace:      namespace,
+			OwnerReference: &ownerReferences,
+		},
+	}
+}
+
+// SetupConfig setups the scheduler config
+func (vsoc *VineyardSchedulerOutsideCluster) SetupConfig() error {
```

Comment:
`vsoc` -> `vs` as there's no `vsok` in `k8s/pkg/schedulers/scheduler_outside_cluster.go` so would be no confusion.